### PR TITLE
Added GraphQL

### PR DESCRIPTION
**Pull request by Neodia**

### CODE REVIEW

File: ui/pages/api/graphql.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { createServer } from "@graphql-yoga/node";
+import { findLectureById } from "../../api/psql_client";
+
+const typeDefs = /* GraphQL */ `
```
Comment (ErnestaP) [inline ui/pages/api/graphql.tsx:4]: do you need this comment?

File: ui/pages/api/graphql.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    files: [String]
+    type: [String]
+  }
+`;
+
+const resolvers = {
+  Query: {
+    async lecture(parent, args, context) {
+      const { id } = args;
+      return findLectureById(id);
```
Comment (ErnestaP) [inline ui/pages/api/graphql.tsx:36]: no need to create a var, you can just pass args.id

File: ui/pages/api/lectures/[id].tsx [hunk 3]
```diff
@@ -1,16 +1,6 @@
...
-    password: "cat",
-  });
-  await client.connect();
-
-  const result = await client.query(
-    "SELECT * FROM cds_lecture WHERE lecture_id = " + id
-  );
-  await client.end();
-  res.status(200).json(result.rows[0]);
+  res.status(200).json(findLectureById(id));
```
Comment (ErnestaP) [inline ui/pages/api/lectures/[id].tsx:5]: would be nice to put error handling, or we don't need it?

File: ui/pages/api/graphql.tsx (hunk 2 above)
Comment (Neodia) [reply ui/pages/api/graphql.tsx]: Done

File: ui/pages/api/lectures/[id].tsx (hunk 3 above)
Comment (Neodia) [reply ui/pages/api/lectures/[id].tsx]: Added

**Pull request merged**